### PR TITLE
[FIX]: GeoFence GET endpoint returns an empty array

### PR DESCRIPTION
**Pull request by cyapa**
## Proposed changes

[JIRA task](https://ssrc.atlassian.net/browse/SCUTM-183)

- Fixed `GET /geo_fence_ops/geo_fence?start_date=XXXX-XX-XX&end_date=2XXXX-XX-XX` not working properly
- Fixed `POST /geo_fence_ops/set_geo_fence` not validating the basic payload correctly.
- Fixed `POST /geo_fence_ops/set_geo_fence` always using the default values for **_GeoFence properties_** instead of the values in JSON payload
- Introduced JSON field validation via Serialisation.
- Added unit tests for `POST /geo_fence_ops/set_geo_fence` endpoint

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply._

- [x] Lint and unit tests pass locally with the changes
- [x] Added new tests that prove the changes in this PR works
- [ ] Added necessary documentation (if appropriate)
- [ ] Added copyrights to new files (if appropriate)

## Further comments

The test JWT we use to run unit tests for the endpoints is being duplicated right now. This will get moved to a common place in the upcoming PRs ❗

### CODE REVIEW

**Review by solita-nikorainto — APPROVED** [review]
💪

File: geo_fence_operations/test_views.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import json
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+JWT = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ0ZXN0ZmxpZ2h0LmZsaWdodGJsZW5kZXIuY29tIiwiY2xpZW50X2lkIjoidXNzX25vYXV0aCIsImV4cCI6MTY4Nzc4Mjk0OCwiaXNzIjoiTm9BdXRoIiwianRpIjoiODI0OWI5ODgtZjlkZi00YmNhLWI2YTctODVhZGFiZjFhMTUwIiwibmJmIjoxNjg3Nzc5MzQ3LCJzY29wZSI6ImJsZW5kZXIucmVhZCIsInN1YiI6InVzc19ub2F1dGgifQ.b63qZWs08Cp1cgfRCtbQfLom6QQyFpqUaFDNZ9ZdAjSM690StACij6FiriSFhOfFiRBv9rE0DePJzElUSwv1r1bI0IpKMtEJYsJY4DXy7ZImiJ3rSten1nnb1LLAELcDIxMZM2D1ek43EFW35al4si640JfMcSmt62bEP1b4Msc"
```
Comment (govindsi) [inline geo_fence_operations/test_views.py:6]: Later probably having .env for UT will make more sense and populate the JWT token from env.

**Review by govindsi — APPROVED** [review]

**Comment by govindsi** [comment]
LGTM, later pls raise upstream PR with out UT.

File: geo_fence_operations/test_views.py (hunk 1 above)
Comment (cyapa) [reply geo_fence_operations/test_views.py:6]: Yeah implementing to have some sort of an efficient way to hold common UT data. Probably with test fixtures or .env. Will look it to this 🙌

**Pull request merged**